### PR TITLE
feat(EMI-1464): fill/clear all inputs when selecting new address

### PR DESCRIPTION
**Pull request by TMcMeans**
The type of this PR is: **FEAT**

This PR solves [EMI-1464]

### Description
Sep-22-2023 11-26-44

[GRO-434]: https://artsyproduct.atlassian.net/browse/GRO-434?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[EMI-1405]: https://artsyproduct.atlassian.net/browse/EMI-1405?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[EMI-1464]: https://artsyproduct.atlassian.net/browse/EMI-1464?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/Components/Address/useAddressAutocomplete.ts [hunk 1]
```diff
@@ -54,6 +57,32 @@ export const useAddressAutocomplete = (
     }
   }, [isUSAddress, result.length])
 
+  // clear address line 2 if selecting a new suggested address
+  useEffect(() => {
+    const newOptions = result.map(
+      (suggestion: ProviderSuggestion): AddressAutocompleteSuggestion => {
+        const text = buildAddressText(suggestion)
+        let newAddressLine2 = suggestion.secondary ? suggestion.secondary : ""
```
Comment (erikdstock) [inline src/Components/Address/useAddressAutocomplete.ts:65]: This can just be suggestion.secondary || ""

Comment (TMcMeans) [reply src/Components/Address/useAddressAutocomplete.ts:65]: Agreed, but I am also exploring removing the useEffect() from this code and seeing how that works out.

**Review by erikdstock — APPROVED** [review]
205943737-5fada7e8-84d4-4e91-ab69-cc6780915537

**Comment by TMcMeans** [comment]
> 205943737-5fada7e8-84d4-4e91-ab69-cc6780915537

I'm both creeped out and impressed by this image... was this done by AI?!

**Pull request merged**